### PR TITLE
Support FluxHelmRelease resources in fluxd

### LINKED ISSUE #985: Release images to FluxChartRelease resources
`FluxChartRelease` resources are effectively workload controllers, or will usually be, since they will typically

 - refer to a chart that includes at least one core workload controller
 - specify particular images and tags (versions) as `values`

To support release automation, the flux machinery will need to understand how to interpret these resources, and how to update them.

### PR DESCRIPTION
**Pull request by squaremo**
This adds support for treating FluxHelmRelease releases as workloads (i.e., things that refer to images and result in pods being created). They will appear in ListImages and ListServices API results, and can be updated and automated like other workloads.

Fixes #985.

### CODE REVIEW

**Comment by squaremo** [comment]
Hmm that scheme of naming the container for the path is not great -- for one thing, it could have slashes, and container names are `dns_label`s I think. Either giving it a hard-wired name (suggestive of the format), or at least using the basename of the path, would be better.

File: cluster/kubernetes/resource/fluxhelmrelease.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+// Containers returns the containers that are defined in the
+// FluxHelmRelease. At present, this assumes only one image in the
+// Spec.Values, which is then named for the chart. If there is no such
+// field, or it is not parseable as an image ref, no containers are
+// returned.
+func (fhr FluxHelmRelease) Containers() []resource.Container {
+	values := fhr.Spec.Values
+	if imgInfo, ok := values["image"]; ok {
+		imgInfoStr := imgInfo.(string)
```
Comment (awh) [inline cluster/kubernetes/resource/fluxhelmrelease.go]: Change this from a type assertion to a test?

**Comment by squaremo** [comment]
> giving it a hard-wired name (suggestive of the format)

I came up with `chart-values-image`. That is suggestive of the format, but only if you know what the format is ... Just `image` would be less mysterious (but still mysterious).

I don't think there's any such name that's going to be entirely self-explanatory. It would be nice to synthesise a name that's meaningful to the chart, but we don't have the chart to hand, and using the path seems unreliable (what if someone supplies `"."` as the chart path?).

**Comment by squaremo** [comment]
> giving it a hard-wired name (suggestive of the format)
> 
> I came up with chart-values-image. That is suggestive of the format, but only if you know what the format is ... Just image would be less mysterious (but still mysterious).

Settled on `chart-image`. Less weirdly specific; indicates how the image value is used. I need to build it into kubeyaml before this will work though.

**Comment by squaremo** [comment]
> I need to build it into kubeyaml before this will work though.

(built into kubeyaml 0.3.1, which is used in the most recent commit)

**Comment by awh** [comment]
:+1:

**Pull request merged**